Source/VCGeneration/Counterexample.cs

```diff
@@ -386,6 +391,113 @@ public Model.Element GetModelValue(Variable v)
     }
 
     public abstract int GetLocation();
+
+    public ErrorInformation CreateErrorInformation(ConditionGeneration.Outcome outcome, bool forceBplErrors)
+    {
+      ErrorInformation errorInfo;
+      var cause = "Error";
+      if (outcome == VCGen.Outcome.TimedOut)
+      {
+        cause = "Timed out on";
+      }
+      else if (outcome == VCGen.Outcome.OutOfMemory)
+      {
+        cause = "Out of memory on";
+      }
+      else if (outcome == VCGen.Outcome.SolverException)
+      {
+        cause = "Solver exception on";
+      }
+      else if (outcome == VCGen.Outcome.OutOfResource)
+      {
+        cause = "Out of resource on";
+      }
+
+      if (this is CallCounterexample callError)
+      {
+        if (callError.FailingRequires.ErrorMessage == null || forceBplErrors)
+        {
+          string msg = callError.FailingCall.ErrorData as string ?? callError.FailingCall.Description.FailureDescription;
+          errorInfo = ErrorInformation.Create(callError.FailingCall.tok, msg, cause);
+          errorInfo.Kind = ErrorKind.Precondition;
+          errorInfo.AddAuxInfo(callError.FailingRequires.tok,
+            callError.FailingRequires.ErrorData as string ?? callError.FailingRequires.Description.FailureDescription,
+            "Related location");
+        }
+        else
+        {
+          errorInfo = ErrorInformation.Create(null, callError.FailingRequires.ErrorMessage, null);
+        }
+      }
+      else if (this is ReturnCounterexample returnError)
+      {
+        if (returnError.FailingEnsures.ErrorMessage == null || forceBplErrors)
+        {
+          errorInfo = ErrorInformation.Create(returnError.FailingReturn.tok, returnError.FailingReturn.Description.FailureDescription, cause);
+          errorInfo.Kind = ErrorKind.Postcondition;
+          errorInfo.AddAuxInfo(returnError.FailingEnsures.tok,
+            returnError.FailingEnsures.ErrorData as string ?? returnError.FailingEnsures.Description.FailureDescription,
```

Comment:
Does this pattern of using `ErrorData as string ?? …` deserve an extension method?

Reply:
There's an issue about removing ErrorData: https://github.com/boogie-org/boogie/issues/562